cpp/include/proton/sender_options.hpp

```diff
@@ -82,7 +82,16 @@ class sender_options {
     /// Set the delivery mode on the sender.
     PN_CPP_EXTERN sender_options& delivery_mode(delivery_mode);
 
-    /// Automatically settle messages (default is true).
+    /**
+     * Automatically settle messages (default is true).
+     *
+     * \note
+     * Sender **auto-settlement** only occurs for a delivery after the
+     * sender receives a settled disposition for that delivery. Otherwise,
+     * there would be no way to receive any further events for that delivery
+     * (such as the subsequent on_settle message that might be expected when
+     * the receiver finally settles the message).
+     */
```

Comment:
I think the suggested wording I made for the python API is good here too (see below). I don't see the need to call out auto-settlement in so many words, just explain the basic behaviour as below.

In the API docs that explain settlement we should explain that you can receive no more events after settling a delivery, but that understanding is fundamental to what *settlement is* and so shouldn't need explaining every place we talk about settlement.

Reply:
Done.